pyfstat/core.py

```diff
@@ -624,19 +626,21 @@ def _get_SFTCatalog(self):
                     )
             C1 = getattr(self, "injectSources", None) is None
             C2 = getattr(self, "injectSqrtSX", None) is None
-            if C1 and C2:
+            C3 = getattr(self, "Tsft", None) is None
+            if (C1 and C2) or C3:
                 raise ValueError(
-                    "If sftfilepattern==None, you must specify either one of"
-                    " injectSources or injectSqrtSX."
+                    "If sftfilepattern==None, you must specify Tsft and"
+                    " either one of injectSources or injectSqrtSX."
                 )
             SFTCatalog = lalpulsar.SFTCatalog()
-            Tsft = 1800
             Toverlap = 0
-            Tspan = self.maxStartTime - self.minStartTime
             self.detector_names = self.detectors.split(",")
             detNames = lal.CreateStringVector(*[d for d in self.detector_names])
+            # MakeMultiTimestamps follows the same [minStartTime,maxStartTime)
+            # convention as the SFT library, so we can pass Tspan like this
```

Comment:
Need to double-check when more awake - the docs [here](https://lscsoft.docs.ligo.org/lalsuite/lalpulsar/group___s_f_tutils__h.html#ga65aa00943b3a912b7dc40ea6213fd3a3) seem to imply the contrary, but I think I ran a check already that seems to imply it's fine...

Reply:
Ok, so the official logic of that functon is:
> the returned list of timestamps are guaranteed to cover the interval [tStart, tStart+duration), assuming a each timestamp covers a length of 'Tsft' This implies that the actual timestamps-coverage can extend up to 'Tsft' beyond 'tStart+duration'. 

(Note that the actual argument is called `Tspan`, not `duration`.)

We pass `Tspan = self.maxStartTime - self.minStartTime` so it should "cover" the interval `[minStartTime,maxStartTime)` which would mean from the `)` notation that `maxStartTime` is **not** included, but from the word `cover` and the following sentence, it **should** be included. If my English reading comprehension isn't failing me, that documentation is self-contradictory. Or at least confusing; it'd be fine if it'd split the ambiguous term `actual timestamps-coverage` into `actual timestamps` which are _only in_ the open interval, and `data coverage` or sth like that which `covers` it. 

And what it actually does:
```python
>>> import lalpulsar
>>> ts=lalpulsar.MakeTimestamps(tStart=700000000,Tspan=1,Tsft=1800,Toverlap=0); print(ts.data)
[LIGOTimeGPS(700000000, 0)]
>>> ts=lalpulsar.MakeTimestamps(tStart=700000000,Tspan=1799,Tsft=1800,Toverlap=0); print(ts.data)
[LIGOTimeGPS(700000000, 0)]
>>> ts=lalpulsar.MakeTimestamps(tStart=700000000,Tspan=1800,Tsft=1800,Toverlap=0); print(ts.data)
[LIGOTimeGPS(700000000, 0)]
>>> ts=lalpulsar.MakeTimestamps(tStart=700000000,Tspan=1801,Tsft=1800,Toverlap=0); print(ts.data)
[LIGOTimeGPS(700000000, 0) LIGOTimeGPS(700001800, 0)]
```
which is consistent with `[tStart, tStart+Tspan)` for the "actual timestamps".

So I'd say it's consistent with our own intentions here and no further PyFstat changes are needed; I'll submit a lalsuite MR rewording the documentation for clarity.

Reply:
We agree this makes sense after discussing/clarifying with @dbkeitel